src/app/admin-section/admin/admin.component.ts

```diff
@@ -38,115 +40,85 @@ export class AdminComponent implements OnInit, OnDestroy {
   ngOnInit() {
     this.userSub = this.authService.user.subscribe(user => {
       this.loggedUser = user;
-
     });
-    this.onGetAllRequests();
-  }
-
-  onGetAllRequests() {
-    this.error = null;
-    this.requestListSub = this.adminService.getAllRequests(this.loggedUser.token)
-      .subscribe(
-        resData => {
-          this.allRequests = resData;
-          if (!this.displayedRequests) {
-            this.displayedRequests = [...this.allRequests]
-          }
-        },
-        error => {
-          if (error.error.error === "unknown user token") {
-            this.error = "Přístup zamítnut, nesprávná autorizace"
-          } else {
-            this.error = "Nelze se připojit k serveru"
-          }
-        }
-      )
+    this.displayedRequestsObservable = this.adminService.getAllRequests(this.loggedUser.token)
+    this.errorObservable = this.adminService.getError()
   }
 
   onApprove(request: oneRequest) {
-    this.error = null;
-    this.requestApproveSub = this.adminService.requestApprove(request, this.loggedUser.token)
-      .subscribe((resData) => {
-          request.status = resData.status;
-        },
-        error => {
-          if (error.error.error === "unknown user token") {
-            this.error = "Přístup zamítnut, nesprávná autorizace"
-          } else {
-            this.error = "Nelze se připojit k serveru"
-          }
-        }
-      )
+    this.adminService.requestApprove(request, this.loggedUser.token)
   }
 
   onReject(request: oneRequest) {
-    this.error = null;
-    this.requestRejectSub = this.adminService.requestReject(request, this.loggedUser.token)
-      .subscribe((resData) => {
-          request.status = resData.status;
-        },
-        error => {
-          if (error.error.error === "unknown user token") {
-            this.error = "Přístup zamítnut, nesprávná autorizace"
-          } else {
-            this.error = "Nelze se připojit k serveru"
-          }
-        }
-      )
+    this.adminService.requestReject(request, this.loggedUser.token)
   }
 
   onFiltering() {
-    let filteredBySubject = this.adminService.displayFilteredSubject(this.allRequests, this.subjectSelection);
-    let filteredByState = this.adminService.displayFilteredState(this.allRequests, this.stateSelection);
-    this.displayedRequests = this.adminService.filterFinal(filteredBySubject, filteredByState);
-    if (this.sortedLoan) {
-      this.onLoanSorting(this.sortedLoan);
-    } else if (this.sortedAlphabet) {
-      this.onAlphabetSorting(this.sortedAlphabet);
-    }
+    this.displayedRequestsObservable = this.displayedRequestsObservable
+      .pipe(
+        map(requests =>
+          this.adminService.displayFilteredSubject(requests, this.subjectSelection)
+        ),
+        map(requests =>
+          this.adminService.displayFilteredState(requests, this.stateSelection)
+        )
+      )
```

Comment:
Jenom takové upozornění, takhle se ti v proměnné displayedRequestsObservable budou neustále nabalovat ty dané filtrace a řazení na sebe, což po dostatečném klikání v aplikaci může způsobit problémy. Řešením by bylo mít jednu proměnnou přímo na tu observable z this.adminService.getAllRequests(this.loggedUser.token) a druhou pro zobrazování